DataCapturing/Persistence/BinarySerializer.swift

```diff
@@ -0,0 +1,299 @@
+//
+//  BinarySerializer.swift
+//  DataCapturing
+//
+//  Created by Team Cyface on 27.02.18.
+//
+
+import Foundation
+import DataCompression
+
+/// The current version of the Cyface binary format.
+let dataFormatVersion: UInt16 = 1
+
+/**
+ Enum used to specify the correct byte order. Apple uses little endian while Cyface uses big endian.
+
+ ````
+ case bigEndian
+ case littleEndian
+ ````
+ */
+enum ByteOrder {
+    /// Big endian byte order. The byte with the highest order is the first.
+    case bigEndian
+    /// little endian byte order. The byte with the lowest order is the first.
+    case littleEndian
+
+    /**
+     Converts the provided value into a byte representation.
+
+     - Parameters:
+     - value: The value to convert to a byte representation.
+     - inOrder: The byte order to use. Apple usually uses little endian, while the Cyface binary format is currently in big endian format, because that is the standard on Android and in the backend system.
+     - Returns: An array of bytes, representing the provided value.
+     */
+    func convertToBytes<T: BinaryInteger>(_ value: T) -> [UInt8] {
+        let ret = value.extractBytes()
+        return self == .bigEndian ? ret.reversed() : ret
+    }
+}
+
+/**
+ Protocol that must be fullfilled by a serializer to transform an object into the Cyface binary format.
+
+ - Author: Klemens Muthmann
+ - Version: 1.0.0
+ - Since: 2.0.0
+ */
+protocol BinarySerializer {
+    /// The type of the item to serialize.
+    associatedtype Serializable
```

Comment:
This is similar to Java Generics for protocols in Swift. Serializable is a placeholder for the actual type to be serialized.